contracts/Pool.sol

```diff
@@ -28,3 +28,79 @@ contract Pool {
         receiver.transfer(value);
     }
 }
+
+struct ReceiverWeight {
+    address next;
+    uint32 weight;
+}
+
+/// @notice Helper methods for receiver weights list.
+/// The list works optimally if after applying a series of changes it's iterated over.
+/// The list uses 1 word of storage per receiver with a non-zero weight.
+library ReceiverWeightsImpl {
+    address internal constant ADDR_ROOT = address(0);
+    address internal constant ADDR_UNINITIALIZED = address(0);
+    address internal constant ADDR_END = address(1);
+
+    /// @notice Return the next non-zero receiver weight and its address.
+    /// Removes all the zeroed items found between the current and the next receivers.
+    /// Iterating over the whole list removes all the zeroed items.
+    /// @param current The previously returned receiver address or zero to start iterating
+    /// @return next The next receiver address
+    /// @return weight The next receiver weight, zero if the end of the list was reached
+    function nextWeight(
+        mapping(address => ReceiverWeight) storage self,
```

Comment:
How come this is passed into each function, vs. being part of the contract global scope?

Reply:
I don't think that it has any gas cost at all, but it lets us have more than one weight mapping. It'll be needed for delegation pools. It also gives us better logic encapsulation.